### PR TITLE
[70218] Add menu node image field

### LINKED ISSUE #114: Image upload option for node
It would be nice to have the option to upload an image to a node. A use case for this would be for a mega menu which contains promotional banners.

This can already be achieved using Cms Block node type, but it can get quite messy to manage, for example if you have a menu with 10 top-level categories, and you want two or three promotional banners within each one. It would be far nicer from a user/content-management perspective if the image could be uploaded directly to the node to avoid having to create separate static blocks.

### PR DESCRIPTION
**Pull request by akai-z**
Resolves #114

### CODE REVIEW

**Review by akai-z — COMMENTED** [review]
TODO: Fix [develop branch merge](https://github.com/SnowdogApps/magento2-menu/pull/140/commits/90a4870bdb588c6b6f5bc7e9be8b89f9bb338101) conflicts in the following files:
[`view/adminhtml/templates/menu/nodes.phtml`](https://github.com/SnowdogApps/magento2-menu/pull/140/commits/90a4870bdb588c6b6f5bc7e9be8b89f9bb338101#diff-9f49d9df8801512b198d845841fc4add2a48a2b43c4c5c3858eb67847466b900)
[`view/adminhtml/web/vue/menu-type.vue`](https://github.com/SnowdogApps/magento2-menu/pull/140/commits/90a4870bdb588c6b6f5bc7e9be8b89f9bb338101#diff-cbb716bce7b489abd95939ddc4983de82ee912bec7f876b2d3e55099a22ec5db)

**Review by emastyle — COMMENTED** [review]
The backend part looks good to me. Vue part still needs to CR by frontend dev and if ok all can be approved.

File: Block/Menu/Edit/Tab/Nodes.php [hunk 1]
```diff
@@ -118,6 +125,30 @@ public function isHidden()
...
+        return $this->getUrl('snowmenu/node/uploadimage');
+    }
+
+    /**
+     * @return string
+     */
+    public function getImageDeleteUrl()
+    {
+        return $this->getUrl('snowmenu/node/deleteimage');
+    }
```
Comment (emastyle) [inline Block/Menu/Edit/Tab/Nodes.php:142]: What do you think if we set the URLs in constants? I.e.:

`const IMAGE_UPLOAD_URL = 'snowmenu/node/uploadimage';`

then in the method:

`public function getImageUploadUrl()
   {
        return $this->getUrl(self::IMAGE_UPLOAD_URL);
   }`

**Review by emastyle — COMMENTED** [review]
The backend code is ok for me, please just check the comment I left for Block/Menu/Edit/Tab/Nodes.php.

File: Block/Menu/Edit/Tab/Nodes.php (hunk 1 above)
Comment (akai-z) [reply Block/Menu/Edit/Tab/Nodes.php]: Yeah, I updated it.

**Review by jochhop — APPROVED** [review]

**Pull request merged**